birdhouse/components/stac/config/magpie/stac_hooks.py

```diff
@@ -0,0 +1,140 @@
+    # type: (Response) -> Response
+    """
+    Create the stac collection resource
+    """
+    request = response.request
+    body = request.json
+    collection_id = body["id"]
+    try:
+        display_name = extract_display_name(body["links"])
+    except Exception as exc:
+        LOGGER.error("Error when extracting display_name from links %s %s", body["links"], str(exc), exc_info=exc)
+        return response
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request, since the current session found in the request is already handled with his own transaction manager.
+        session = get_session_from_other(request.db)
+        # Create the resource tree
+        create_resource_tree(f"stac/collections/{collection_id}", 0, service.resource_id , session, display_name)
+        session.commit()
+
+    except Exception as exc:
+        LOGGER.error("Unexpected error while creating the collection %s %s", display_name, str(exc), exc_info=exc)
+        session.rollback() 
+        return response
+
+    return response
+
+def create_item_resource(response):
+    # type: (Response) -> Response
+    """
+    Create the stac item resource
+    """
+    request = response.request
+    body = request.json
+    item_id = body["id"]
+    try:
+        display_name = extract_display_name(body["links"])
+    except Exception as exc:
+        LOGGER.error("Error when extracting display_name from links %s %s", body["links"], str(exc), exc_info=exc)
+        return response
+
+    # Get the <collection_id> from url -> /collections/{collection_id}/items
+    collection_id = re.search(r'(?<=collections\/).*?(?=\/items)', request.url).group()
+
+    # note: matchdict reference of Twitcher owsproxy view is used, just so happens to be same name as Magpie
+    service = get_service_matchdict_checked(request)
+    try:
+        # Getting a new session from the request, since the current session found in the request is already handled with his own transaction manager.
+        session = get_session_from_other(request.db)
+        # Create the resource tree
+        create_resource_tree(f"stac/collections/{collection_id}/items/{item_id}", 0, service.resource_id, session, display_name)
+        session.commit()
+
+    except Exception as exc:
+        LOGGER.error("Unexpected error while creating the item %s %s", display_name, str(exc), exc_info=exc)
+        session.rollback()
+        return response
+
+    return response
+
+def extract_display_name(links):
+    # type: (List[Dict[str, str]]) -> str
+    """
+    Extract THREDD path from a STAC links
+    """
+    display_name = None
+    for link in links:
+        if link["rel"] == "source":
+            # Example of title `thredds:birdhouse/CMIP6` -> `birdhouse/CMIP6`
+            display_name = link["title"].split(":")[1]
```

Comment:
Shouldn't the service part be kept?
Otherwise, how will Cowbird know under which service the associated resource is located under?

Reply:
In the cowbird config? If we leave "thredds:" in the display_name, won't we need to split it in cowbird before being able to map the directory path to a stac item ?

Reply:
You should be able to define a regex in the cowbird path segments. 
Provided that the feature that allows extracting `resource_display_name` instead of `resource_name` is added, it should be possible to have `thredds:{var}`. Maybe some adjustments must be added in https://github.com/Ouranosinc/cowbird/blob/aad19bac737bc0ff60bb3c2cf977eeb9383c9f85/cowbird/permissions_synchronizer.py for `_generate_regex_from_segments` and `_find_matching_res`. 
The `thredds:` prefix would be important in order to support multiple distinct THREDDS servers.